hilti/toolchain/include/ast/operators/enum.h

```diff
@@ -35,4 +35,17 @@ Instantiates an enum instance initialized from an integer value. The value does
     }
 END_CTOR
 
+BEGIN_METHOD(enum_, HasLabel)
+    auto signature() const {
+        return Signature{.self = type::constant(type::Enum(type::Wildcard())),
+                         .result = type::Bool(),
+                         .id = "has_label",
+                         .args = {},
+                         .doc = R"(
+Returns *true* if the value of *op1* corresponds to a known enum label (other
```

Comment:
At least to me "label" in the context of enums doesn't immediately ring a bell, and I could imagine that this could be hard to discover for users. What do you think about instead using something like `is_know`? That should capture both the cases of uninitialized enum values and enum values filled with values outside of the expected range.

Reply:
On https://docs.zeek.org/projects/spicy/en/latest/programming/language/types.html
in 5.2.5.5, the use of the word label is widespread and consistent for this concept, i.e.

Enum types associate labels with a numerical values.
(I have just reported that grammar error, over in the docs pinned Issue)

The only other uses of label to mean something different are in 5.2.5.2
Each RANGE has one of the forms LABEL: A or LABEL: A..B where A and B are a bit numbers.
(should be
Each RANGE has one of the forms LABEL: A or LABEL: A..B where A and B are bit numbers.
I reported that grammar error too, over in the docs pinned Issue)

and one use of the word label in 5.2.5.21
`...you can initialize unit instances through coercion from a list expression: my_unit = [$FIELD_1 = X_1, $FIELD_N = X_N, ...] where FIELD_I is the label of a corresponding field in my_unit’s type.`

The uses of the word label in 5.2.5.2 and 5.2.5.21 can probably change (perhaps to: name of the member).

Reply:
Yeah, I had been looking at 5.2.5.5 when picking the name, so it is consistent with the docs. We also use "label" at other locations with enums in the docs. Google also shows "label" being used with enums in a various contexts. May not be the most common terminology, but there's precedent. :-) So, I'll leave as is.